docs/events.md

```diff
@@ -4,63 +4,60 @@ linkTitle: "Events"
+   if it cannot execute at all due to failing validation.
+
+## Events in `PipelineRuns`
+
+`PipelineRuns` emit events for the following `Reasons`:
+
+- `Started`: emitted the first time the `PipelineRun` is picked by the
   reconciler from its work queue, so it only happens if web-hook validation was
-  successful. Note that this event does not imply that a step started executing,
-  as pipeline, task and bound resource validation must be successful first.
-- `Running`: this is triggered when the `PipelineRun` passes validation and
-  actually starts running.
-- `Succeeded`: this is triggered once all `Tasks` reachable via the DAG are
+  successful. This event in itself does not indicate that a `Step` is executing;
+  the `Step` executes once validation for the `Pipeline` as well as all associated `Tasks`
+  and `Resources` is successful.
+- `Running`: emitted when the `PipelineRun` passes validation and
+  actually begins execution.
+- `Succeeded`: emitted once all `Tasks` reachable via the DAG have
   executed successfully.
-- `Failed`: this is triggered if the `PipelineRun` is completed, but not
-  successfully. Causes of failure may be: one the `Tasks` failed or the
-  `PipelineRun` was cancelled or timed out. `Failed` events are also triggered
-  in case the `PipelineRun` cannot be executed at all because of validation issues.
-
-# CloudEvents
-
-When a sink is [configured](./install.md#configuring-cloudevents-notifications), the following events
-will be generated by appropriate controller when a lifecycle event happens for  `TaskRun`.
-
-The complete table of events:
-
-Reasource     |Event    |Event Type
-:-------------|:-------:|:----------------------------------------------------------
-TaskRun       | Started | dev.tekton.event.taskrun.started.v1
-TaskRun       | Running | dev.tekton.event.taskrun.runnning.v1
-TaskRun       | Condition Change while Running | dev.tekton.event.taskrun.unknown.v1
-TaskRun       | Succeed | dev.tekton.event.taskrun.successful.v1
-TaskRun       | Failed  | dev.tekton.event.taskrun.failed.v1
+- `Failed`: emitted if the `PipelineRun` finishes running unsuccessfully because a `Task` failed or the
+  `PipelineRun` timed out or was cancelled. A `PipelineRun` also emits `Failed` events if it cannot
+  execute at all due to failing validation.
+
+# Events via `CloudEvents`
+
+When you [configure a sink](install.md#configuring-cloudevents-notifications), Tekton emits
+events as described in the table below.
+
+Resource  |Event    |Event Type
+:---------|:-------:|:----------------------------------------------------------
+`TaskRun` | `Started` | `dev.tekton.event.taskrun.started.v1`
+`TaskRun` | `Running` | `dev.tekton.event.taskrun.runnning.v1`
+`TaskRun` | `Condition Change while Running` | `dev.tekton.event.taskrun.unknown.v1`
+`TaskRun` | `Succeed` | `dev.tekton.event.taskrun.successful.v1`
+`TaskRun` | `Failed` | `dev.tekton.event.taskrun.failed.v1`
```

Comment:
@afrittoli do we create an event for pipeline `stopping`? I believe we do, please confirm.

Reply:
There's no event dedicated to 'stopping'.
Any reason other than "Running" with status "Unknown" triggers the generic event `dev.tekton.event.taskrun.unknown.v1`:

```
	case c.IsUnknown():
		switch runObject.(type) {
		case *v1beta1.TaskRun:
			switch c.Reason {
			case v1beta1.TaskRunReasonStarted.String():
				eventType = TaskRunStartedEventV1
			case v1beta1.TaskRunReasonRunning.String():
				eventType = TaskRunRunningEventV1
			default:
				eventType = TaskRunUnknownEventV1
			}
```

We could add one though :)

Reply:
If we decide to move forward with adding the "Stopping" event, let's update this doc again after it's added instead of holding this PR up?

Reply:
yeah, definitely not something for this PR :)

Reply:
SGTM, resolving the conversation for now.

Reply:
If its not already generated, I wouldnt create one now, its something internal to pipeline controller and would be better to keep it that way 🙏

Reply:
I was referring to `PipelineRun` event for `stopping`, sorry added my comment in the wrong section 😞 

This PR looks good, no additional changes needed here and its merged.

https://github.com/tektoncd/pipeline/blob/0b345b56bda47c4542519102527889f7d0777a33/pkg/reconciler/pipelinerun/pipelinerun.go#L462-L470

@afrittoli Events are created with the reason in them and reason is set to `Stopping` here: 🤔 

https://github.com/tektoncd/pipeline/blob/8e9230d68634e9426a0503ec3dcc2a7a877c04cc/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L581-L585